src/databricks/labs/remorph/connections/database_manager.py

```diff
@@ -0,0 +1,71 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine, Result
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import text
+
+
+class _ISourceSystemConnector(ABC):
+    @abstractmethod
+    def _connect(self) -> Engine:
+        pass
+
+    @abstractmethod
+    def execute_query(self, query: str) -> Result[Any]:
+        pass
+
+
+class _BaseConnector(_ISourceSystemConnector):
+    def __init__(self, config: dict[str, Any]):
+        self.config = config
+        self.engine: Engine = self._connect()
+
+    def _connect(self) -> Engine:
+        raise NotImplementedError("Subclasses should implement this method")
+
+    def execute_query(self, query: str) -> Result[Any]:
+        if not self.engine:
+            raise ConnectionError("Not connected to the database.")
+        session = sessionmaker(bind=self.engine)
+        connection = session()
+        return connection.execute(text(query))
+
+
+def _create_connector(db_type: str, config: dict[str, Any]) -> _ISourceSystemConnector:
+    connectors = {
+        "snowflake": SnowflakeConnector,
```

Comment:
Why have `snowflake` since we don't yet support it ?

Reply:
I was going for placeholder we can definitely remove them.